apps/api/src/app/inbox/usecases/notifications-count/notifications-count.usecase.ts

```diff
@@ -0,0 +1,80 @@
+import { Injectable } from '@nestjs/common';
+import { MessageRepository, SubscriberRepository } from '@novu/dal';
+import { ChannelTypeEnum, MessagesStatusEnum } from '@novu/shared';
+import { buildMessageCountKey, CachedQuery } from '@novu/application-generic';
+
+import { NotificationsCountCommand } from './notifications-count.command';
+import { ApiException } from '../../../shared/exceptions/api.exception';
+
+const MAX_NOTIFICATIONS_COUNT = 100;
+
+@Injectable()
+export class NotificationsCount {
+  constructor(private messageRepository: MessageRepository, private subscriberRepository: SubscriberRepository) {}
+
+  private constuctQuery(command: NotificationsCountCommand) {
+    const uniqueStatuses = new Set(command.status);
+    const query = [...uniqueStatuses.values()].reduce<{ feedId?: string[]; seen?: boolean; read?: boolean }>(
+      (acc, el) => {
+        let seen: boolean | undefined;
+        if ([MessagesStatusEnum.SEEN, MessagesStatusEnum.UNSEEN].includes(el)) {
+          seen = typeof acc.seen === 'undefined' ? el === MessagesStatusEnum.SEEN : undefined;
+
+          return {
+            ...acc,
+            seen,
+          };
+        }
+
+        let read: boolean | undefined;
+        if ([MessagesStatusEnum.READ, MessagesStatusEnum.UNREAD].includes(el)) {
+          read = typeof acc.read === 'undefined' ? el === MessagesStatusEnum.READ : undefined;
+
+          return {
+            ...acc,
+            read,
+          };
+        }
+
+        return acc;
+      },
+      {
+        feedId: command.feedId,
+      }
+    );
+
+    return query;
+  }
```

Comment:
constructs the MongoDB query based on the notification statuses array, for example:
```
let statuses = [MessagesStatusEnum.SEEN, MessagesStatusEnum.READ];
let query = constuctQuery({ statuses, ... });
// query --> { seen: true, read: true }

statuses = [MessagesStatusEnum.SEEN, MessagesStatusEnum.UNSEEN];
query = constuctQuery({ statuses });
// query --> { seen: undefined },  which means in the DAL layer to get all seen in [true, false]
```

In the future, it will be easy to introduce `MessagesStatusEnum.ARCHIVE`.

Reply:
Nice. It would be nice to move this to a utility helper.